### ISSUE TITLE
How to add multi-res icons to an AppImage

### ISSUE THREAD

**Issue by Skywalker13** [AppImage/AppImageKit#438]
Hello,

based on https://github.com/AppImage/AppImageKit/wiki/Creating-AppImages#manually-creating-an-appdir, we can put the icon alongside the desktop file in the `AppDir`.

```
MyApp.AppDir/
MyApp.AppDir/AppRun
MyApp.AppDir/myapp.desktop
MyApp.AppDir/myapp.png
MyApp.AppDir/usr/bin/myapp
MyApp.AppDir/usr/lib/libfoo.so.0
```

It's possible to put an `.svg` instead of the `.png` and it works fine.. but it's not enough.

I have low-resolution icons and a SVG. For example I would like to put PNG in
```
$HOME/.local/share/icons/hicolor/16x16/apps
$HOME/.local/share/icons/hicolor/32x32/apps
```
and SVG in
```
$HOME/.local/share/icons/hicolor/scalable/apps
```

How to do that please?

**Comment by TheAssassin** [comment]
I'd say, go for the SVG and discard your PNG. The PNG can be scales down by your application anyway.

Our tools will use the first icon they find anyway, they won't use multiple icons, which means that the files would just be bundled but not used by anything.

The way it works: the tools parse your desktop file, check for the icon name, look for a matching one, copy it to the AppDir root and use it from there. This means that in many cases there is a dupe in usr/something. As I recently implemented an exclude file feature, appimagetool will soon start to exclude the dupes. But that's actually off topic.

TL;DR: Use the SVG.

**Comment by Skywalker13** [comment]
OK, I can live with it even if the SVG has too much details for low res (and looks ugly in this case)

**Comment by TheAssassin** [comment]
You could always improve the SVG to fit your needs. But on Hi DPI devices an SVG will always fit, a low res PNG looks ugly though.

As usual, you decide what goes in, so you're not bound to my suggestion in any way.

**Comment by probonopd** [comment]
Well @Skywalker13, if you put your icons in the usual `usr/share/icons/hicolor/...x.../apps/...` hierarchy inside the AppImage, then the optional `appimaged` daemon will register all of them with the system.

**Comment by Skywalker13** [comment]
Great, thank you

**Comment by TheAssassin** [comment]
> Well @Skywalker13, if you put your icons in the usual usr/share/icons/hicolor/...x.../apps/... hierarchy inside the AppImage, then the optional appimaged daemon will register all of them with the system.

Didn't know that. Is there any reason for linuxdeployqt to copy the icon file to the AppDir root then? I really doubt that this is necessary, as appimaged is the only application that actually uses the icon.

**Comment by probonopd** [comment]
The icon file to the AppDir root is the one that must be there. The stuff in `usr/share/icons` is optional and only used by `appimaged` for desktop integration.

**Comment by TheAssassin** [comment]
As linuxdeployqt has issues when the file is not in `usr/` though but creates a dupe if it uses one from there, how about having it create a symlink instead of creating an actual copy instead?

**Comment by probonopd** [comment]
I don't think all tools (e.g., `appimaged`) can already handle symlinks. Shame on me...

**Comment by probonopd** [comment]
squashfs deduplicates though :)

**Comment by TheAssassin** [comment]
Oh, okay, then never mind. Although that's just a workaround, I think that's sufficient.
